### PR TITLE
[Feature] Add CharRecallPrecision for OCR Task

### PR DESCRIPTION
**Pull request by Harold-lkk**
Adapt at https://github.com/open-mmlab/mmocr/pull/1728

### CODE REVIEW

File: mmeval/metrics/char_recall_precision.py [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+            - unchanged: Do not change prediction texts and labels.
+            - upper: Convert prediction texts and labels into uppercase
+                     characters.
+            - lower: Convert prediction texts and labels into lowercase
+                     characters.
+            Usually, it only works for English characters. Defaults to
+            'unchanged'.
+        valid_symbol (str): A regular expression to filter out invalid or
+            not cared characters. Defaults to '[^A-Z^a-z^0-9^\u4e00-\u9fa5]'.
+        **kwargs: Keyword parameters passed to :class:`BaseMetric`.
```
Comment (C1rN09) [inline mmeval/metrics/char_recall_precision.py:23]:
```suggestion
    Args:
        letter_case (str): There are three options to alter the letter cases
        
            - unchanged: Do not change prediction texts and labels.
            - upper: Convert prediction texts and labels into uppercase
               characters.
            - lower: Convert prediction texts and labels into lowercase
               characters.
               
            Usually, it only works for English characters. Defaults to
            'unchanged'.
        valid_symbol (str): A regular expression to filter out invalid or
            not cared characters. Defaults to '[^A-Z^a-z^0-9^\u4e00-\u9fa5]'.
        **kwargs: Keyword parameters passed to :class:`BaseMetric`.
```

File: mmeval/metrics/char_recall_precision.py [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+        letter_case (str): There are three options to alter the letter cases
+            - unchanged: Do not change prediction texts and labels.
+            - upper: Convert prediction texts and labels into uppercase
+                     characters.
+            - lower: Convert prediction texts and labels into lowercase
+                     characters.
+            Usually, it only works for English characters. Defaults to
+            'unchanged'.
+        valid_symbol (str): A regular expression to filter out invalid or
+            not cared characters. Defaults to '[^A-Z^a-z^0-9^\u4e00-\u9fa5]'.
```
Comment (C1rN09) [inline mmeval/metrics/char_recall_precision.py:22]: `\u4e00-\u9fa5` renders to strange symbols on api docs. Shourd we add a 'r' at the beginging of this docstring?

File: mmeval/metrics/char_recall_precision.py [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+        for gt, pred, true_positive in results:
+            gt_sum += gt
+            pred_sum += pred
+            true_positive_sum += true_positive
+        char_recall = true_positive_sum / max(gt_sum, 1.0)
+        char_precision = true_positive_sum / max(pred_sum, 1.0)
+        eval_res = {}
+        eval_res['recall'] = char_recall
+        eval_res['precision'] = char_precision
+        return eval_res
```
Comment (C1rN09) [inline mmeval/metrics/char_recall_precision.py:85]:
```suggestion
        metric_results = {}
        metric_results['recall'] = char_recall
        metric_results['precision'] = char_precision
        return metric_results
```

**Review by C1rN09 — APPROVED** [review]
LGTM. Should rebase to `main` branch to fix CI

**Review by zhouzaida — APPROVED** [review]

**Pull request merged**